src/reports/models/reportsManager.js

```diff
@@ -88,11 +92,30 @@ module.exports.postStats = async (report, stats) => {
     }
     await databaseConnector.updateReport(report.test_id, report.report_id, { phase: report.phase, last_updated_at: statsTime });
     report = await module.exports.getReport(report.test_id, report.report_id);
+    await updateReportBenchMarkIfNeeded(report);
     notifier.notifyIfNeeded(report, stats);
 
     return stats;
 };
 
+async function updateReportBenchMarkIfNeeded(report) {
+    if (!reportUtil.isAllRunnersInExpectedPhase(report, constants.SUBSCRIBER_DONE_STAGE)) {
+        return;
+    }
+    const testBenchMarkData = await extractBenchMarkData(report.test_id);
+    if (testBenchMarkData) {
+        const reportAggregate = await aggregateReportManager.aggregateReport(report);
+        const reportBenchMark = benchmarkCalculator.calculate(testBenchMarkData, reportAggregate.aggregate);
+        const { data, score } = reportBenchMark;
```

Comment:
benchmarkCalculator.calculate returns data, but data is the input
need to change that.

i would suggest the function
benchmarkCalculator.calculate will be called benchmarkCalculator.calculateReportScore and return number

Reply:
it was only temp, it will return the score, and data is the way we got to this score it will not be the same data as input